test/augmentation/test_augmentation.py

```diff
@@ -25,6 +25,334 @@
+        assert_allclose(f(input)[1], expected_transform)
+
+
+class TestColorJitterAlternative(CommonTests):
+
+    possible_params = {
+        "brightness": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "contrast": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "saturation": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "hue": (0., (-0.5, 0.5), [-0.5, 0.5], torch.tensor(0.), torch.tensor((-0.5, 0.5))),
+    }
+    params = [{param_name: v} for param_name, possible_values in possible_params.items() for v in possible_values]
+
+    _augmentation_cls = ColorJitter
+    _default_param_set = {"brightness": 0.3,
+                          "contrast": 0.3,
+                          "saturation": 0.3,
+                          "hue": 0.3, }
+
+    @pytest.fixture(params=params, scope="class")
+    def param_set(self, request):
+        return request.param
+
+    @pytest.mark.parametrize("input_shape", [(3, 4, 5), (2, 3, 4, 5)])
+    def test_consistent_output_shape(self, device, dtype, input_shape):
+        self._test_consistent_output_shape_implementation(
+            device=device, dtype=dtype, input_shape=input_shape, **self._default_param_set)
+
+    def test_random_p_1(self, device, dtype):
+        torch.manual_seed(42)
+        # fmt: off
+        input_tensor = torch.tensor([[.1, .2, .3, .4],
+                                     [.5, .6, .7, .8],
+                                     [.9, .0, .1, .2]], device=device, dtype=dtype).repeat(3, 1, 1)
+        expected_output = torch.tensor([[[[0.6024, 0.7273, 0.8522, 0.9771],
+                                          [1.0000, 1.0000, 1.0000, 1.0000],
+                                          [1.0000, 0.4775, 0.6024, 0.7273]]]]).repeat(1, 3, 1, 1)
+        # fmt: on
+        self._test_random_p_1_implementation(device=device, dtype=dtype, input_tensor=input_tensor, expected_output=expected_output,
+                                             brightness=0.5, contrast=0.3, saturation=[0.2, 1.2], hue=0.1
+                                             )
+
+    def test_random_p_1_return_transform(self, device, dtype):
+        torch.manual_seed(42)
+        # fmt: off
+        input_tensor = torch.tensor([[.1, .2, .3, .4],
+                                     [.5, .6, .7, .8],
+                                     [.9, .0, .1, .2]], device=device, dtype=dtype).repeat(3, 1, 1)
+        expected_output = torch.tensor([[[[0.6024, 0.7273, 0.8522, 0.9771],
+                                          [1.0000, 1.0000, 1.0000, 1.0000],
+                                          [1.0000, 0.4775, 0.6024, 0.7273]]]], device=device, dtype=dtype).repeat(1, 3, 1, 1)
+        expected_transformation = torch.tensor([[[1., 0., 0.],
+                                                 [0., 1., 0.],
+                                                 [0., 0., 1.]]], device=device, dtype=dtype)
```

Comment:
can we take here inputs to `_test_random_p_1_return_transform_implementation` with decorators to make it clean ? /cc @anguelos

Reply:
Hmm I 'm not sure if I understand what you mean.